pkg/sql/schemachanger/sctest/comparator.go

```diff
@@ -74,25 +74,22 @@ func CompareLegacyAndDeclarative(t *testing.T, ss StmtLineReader) {
 
 	for ss.HasNextLine() {
 		line := ss.NextLine()
+		declarativeLine := line
 		syntaxError := hasSyntaxError(line)
 
 		// Pre-execution: modify `line` so that executing it produces the same
-		// state. This step is to account for the known behavior difference between
-		// the two schema changers.
-		// Only run when not in a transaction (otherwise certain DDL combo can make
-		// sql queries issued during modification break; see commit message) and
-		// `line` is single DDL statement.
-		if !isInATransaction(ctx, t, legacyConn) && !syntaxError && singleDDLStmt(line) {
-			line = modifyBlacklistedStmt(ctx, t, line, legacyConn)
-		}
+		// descriptor state. This step is primarily for accounting for the known
+		// behavior difference between the two schema changers.
+		line, declarativeLine = preExecutionProcessing(ctx, t, line, legacyConn, declarativeConn)
```

Comment:
This is starting to get pretty complex and difficult to follow as a 3rd party. It may be worthwhile to add in a struct to help wrap up all these variables and provide a convenient place for documentation.

```go
// Line represents a single SQL (line | statement) that is to be executed and cross examined across the legacy and declarative schema changers
type Line struct {
Original string

// Legacy is a copy of Original, potentially with modifications, for execution on the Legacy Schema Changer.
Legacy string

// You get the idea ...
Declarative string
}
```